torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -4064,6 +4067,42 @@ Tensor linalg_det_backward(
   }
 }
 
+std::tuple<Tensor, Tensor, Tensor> attn_backward(
+    const Tensor& grad_o,
+    const Tensor& grad_a,
+    const Tensor& q,
+    const Tensor& k,
+    const Tensor& v) {
+  if (!grad_o.defined() && !grad_a.defined()) {
+    return {};
+  }
```

Comment:
Whenever all the grad inputs are undefined, the engine should just won't compute the backward for this function at all, so technically we don't need this check.

Reply:
oh